zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,875 @@
+        Either a new dataset or bound column based on the shape of the expr
+        passed in. If a table shaped expression is passed, this will return
+        a ``DataSet`` that represents the whole table. If an array-like shape
+        is passed, a ``BoundColumn`` on the dataset that would be constructed
+        from passing the parent is returned.
+    """
+    deltas = _get_deltas(expr, deltas, no_deltas_rule)
+    if deltas is not None:
+        invalid_nodes = tuple(filter(is_invalid_deltas_node, expr._subterms()))
+        if invalid_nodes:
+            raise TypeError(
+                'expression with deltas may only contain (%s) nodes,'
+                " found: %s" % (
+                    ', '.join(map(getname, valid_deltas_node_types)),
+                    ', '.join(set(map(compose(getname, type), invalid_nodes))),
+                ),
+            )
+
+    # Check if this is a single column out of a dataset.
+    if bz.ndim(expr) != 1:
+        raise TypeError(
+            'expression was not tabular or array-like,'
+            ' %s dimensions: %d' % (
+                'too many' if bz.ndim(expr) > 1 else 'not enough',
+                bz.ndim(expr),
+            ),
+        )
+
+    single_column = None
+    if isscalar(expr.dshape.measure):
+        # This is a single column. Record which column we are to return
+        # but create the entire dataset.
+        single_column = rename = expr._name
+        field_hit = False
+        if not isinstance(expr, traversable_nodes):
+            raise TypeError(
+                "expression '%s' was array-like but not a simple field of"
+                " some larger table" % str(expr),
+            )
+        while isinstance(expr, traversable_nodes):
+            if isinstance(expr, bz.expr.Field):
+                if not field_hit:
+                    field_hit = True
+                else:
+                    break
+            rename = expr._name
+            expr = expr._child
+        expr = expr.relabel({rename: single_column})
+
+    measure = expr.dshape.measure
+    if not isrecord(measure) or AD_FIELD_NAME not in measure.names:
+        raise TypeError(
+            "expr must be a collection of records with at least an '{ad}'"
+            " field. Fields provided: '{fields}'\nhint: maybe you need to use "
+            " `relabel` to change your field names".format(
+                ad=AD_FIELD_NAME,
+                fields=measure,
+            ),
+        )
+    _check_datetime_field(AD_FIELD_NAME, measure)
+
+    if TS_FIELD_NAME not in measure.names:
+        expr = bz.transform(expr, **{TS_FIELD_NAME: expr[AD_FIELD_NAME]})
+        if deltas is not None:
+            deltas = bz.transform(
+                deltas,
+                **{TS_FIELD_NAME: deltas[AD_FIELD_NAME]}
+            )
+    else:
+        _check_datetime_field(TS_FIELD_NAME, measure)
+
+    if deltas is not None and (sorted(deltas.dshape.measure.fields) !=
```

Comment:
Is fields just name of the fields?  Will this catch the case of two expressions with the same field names but different dtypes (which I assume we can't sanely support)?